craft_application/util/cli.py

```diff
@@ -0,0 +1,37 @@
+#  This file is part of craft-application.
+#
+#  Copyright 2024 Canonical Ltd.
+#
+#  This program is free software: you can redistribute it and/or modify it
+#  under the terms of the GNU Lesser General Public License version 3, as
+#  published by the Free Software Foundation.
+#
+#  This program is distributed in the hope that it will be useful, but WITHOUT
+#  ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+#  SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.
+#  See the GNU Lesser General Public License for more details.
+#
+#  You should have received a copy of the GNU Lesser General Public License
+#  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""CLI-related utilities."""
+
+import datetime
+
+
+def format_timestamp(dt: datetime.datetime) -> str:
+    """Convert a datetime object (with or without timezone) to a string.
+
+    The format is an ISO 8601-compliant UTC date and time stamp formatted as:
+
+        <DATE>T<TIME>Z
+
+    Timezone-aware datetime objects are converted to UTC. Timezone-naive ones
+    are assumed to be UTC.
+    """
+    if dt.tzinfo is not None and dt.tzinfo.utcoffset(None) is not None:
+        # timezone aware
+        dtz = dt.astimezone(datetime.timezone.utc)
+    else:
+        # timezone naive, assume it's UTC
+        dtz = dt
```

Comment:
I wonder if we should emit a warning here, since we should never be using naive datetimes if we can avoid it?

```suggestion
        dtz = dt
        warnings.warn("Timezone-naive datetime used. Replace with a timezone-aware one if possible.", stacklevel=2)
```

Reply:
Added in https://github.com/canonical/craft-application/pull/599/commits/70eb7d02e8faae4f7618846e36923181106a7af3